src/qibosoq/components/base.py

```diff
@@ -75,9 +77,9 @@ class Sweeper:
     """Number of points of the sweeper."""
     parameters: List[Parameter]
     """List of parameter to update."""
-    starts: List[Union[int, float]]
+    starts: Union[List[float], np.ndarray]
```

Comment:
Isn't it possible to support only `npt.NDArray`? (dropping lists)

Reply:
Maybe yes, but at the cost of more complexity in the serialization/deserialization for the communication...
I could add a dict_factory to use the asdict to convert arrays into list and then change the __init__ to accept also lists for the serialization.
This will also change the "logic" of the commands dictionary that will not be anymore a strict dictionary representation of the objects...

Not sure if it's much better than now

Reply:
> Maybe yes, but at the cost of more complexity in the serialization/deserialization for the communication...

More or less that was the proposal above: relegate all the complexity to the serialization/deserialization, and after that you could rely on them being arrays, avoid any further conversion.

So, the idea is that when you have the object, it contains arrays (which are other objects), and when you serialize, the object becomes a dict, and the arrays lists.

Reply:
With these lines I am sure that they are arrays everywhere on the server:

Reply:
On the client side, I would actually construct either everything in the serialized representation, or everything in the object fashion (using `qibosoq` classes and methods).

So, I'd clearly separate the two worlds: serialized and deserialized. And then, make tools to make the two of them equivalent.
What I wouldn't do is to blend the two worlds, because then you add further conversions, you need to take into account more cases, and sometimes you also need to take manual steps before converting *to the other side* (i.e. if you are in the middle, but you only have a map from one edge to the other and conversely, you might have to move to the closest edge before being able to jump to the other).

Reply:
I have now removed the support for lists in starts and stops, I left it for indexes and parameters...
Should I also remove it from them? 
Actually I tried, but had some problem with the numpy.array of Parameter objects.  
It shouldn't matter much anyway, I suppose...

Reply:
I exposed above the general principle, but do not worry too much about getting everything perfect in just one shot.

The best fix in this case, if everything still works, and only Mypy is complaining, just make the workflow succeed anyhow, ignoring the outcome of Mypy.
Whenever we will solve all Mypy problems, we could drop the toggle and allow workflow failures.

Reply:
Actually the problem was not only of mypy, but things were not working at all... But I think that, as it is now, it's good enough :-)